napari/layers/points/points.py

```diff
@@ -336,13 +359,41 @@ def __init__(
 
         self.refresh_colors()
 
-        # set the current_* properties
-        self._current_edge_color = self.edge_color[-1]
-        self._current_face_color = self.face_color[-1]
         self.size = size
-        self.current_properties = {
-            k: np.asarray([v[-1]]) for k, v in self.properties.items()
-        }
+        # set the current_* properties
+        if len(data) > 0:
+            self._current_edge_color = self.edge_color[-1]
+            self._current_face_color = self.face_color[-1]
+            self.current_properties = {
+                k: np.asarray([v[-1]]) for k, v in self.properties.items()
+            }
+        elif len(data) == 0 and self.properties:
+            self.current_properties = {
+                k: np.asarray([v[0]])
+                for k, v in self.default_properties.items()
+            }
+            if self._edge_color_mode == ColorMode.DIRECT:
+                self._current_edge_color = transform_color_with_defaults(
+                    num_entries=1,
+                    colors=edge_color,
+                    elem_name="edge_color",
+                    default="white",
+                )
+            else:
+                self._current_edge_color = np.array([0, 0, 0, 1])
+            if self._face_color_mode == ColorMode.DIRECT:
+                self._current_face_color = transform_color_with_defaults(
+                    num_entries=1,
+                    colors=face_color,
+                    elem_name="face_color",
+                    default="white",
+                )
+            else:
+                self._current_face_color = np.array([1, 1, 1, 1])
```

Comment:
I don't understand why this is what happens when the color mode is *not* direct...?

Reply:
@kevinyamauchi did you see this question?

Reply:
Ah, sorry about that - missed this!

This situation is that we need to initialize with a `_current_*_color` value and when we have an empty layer where the colors are mapped to a property, we don't really have a "current" color. I had initially given a placeholder default value during development and forgot to circle back to fix it. I think we can now give it the color that would correspond to the default property value from `property_choices`